packages/common/workspace/src/engine/blob.ts

```diff
@@ -78,10 +95,16 @@ export class BlobEngine {
               await remote.set(key, data);
             }
           } catch (err) {
-            logger.error(
-              `error when sync ${key} from [${this.local.name}] to [${remote.name}]`,
-              err
-            );
+            const code = (err as any)[0].extensions.code;
```

Comment:
```suggestion
            if (err instanceof GraphQLError) {
            const code = (err as any)[0].extensions.code;
```

Reply:
This part of the error check is only for affine cloud, while BloeEngine is a general class

we can define a `BlobStorageOverCapacity` error,

Then check the status code === 413 in `affineCloudBlobStorage`, and then throw a `BlobStorageOverCapacity`.

```ts
if (err instanceof BlobStorageOverCapacity) {
  this.isOverCapacity = true
}
```

Reply:
How does it look now? Does it match your idea?